ambry-commons/src/test/java/com.github.ambry.commons/ByteBufferReadableStreamChannelTest.java

```diff
@@ -177,10 +180,75 @@ public void idempotentOperationsTest()
     assertFalse("ByteBufferReadableStreamChannel is not closed", byteBufferReadableStreamChannel.isOpen());
   }
 
+  /**
+   * Tests construction of {@link ByteBufferReadableStreamChannel} on some corner cases.
+   * <p/>
+   * Corner case list:
+   * 1. Create with a null {@link ByteBuffer}.
+   * 2. Create with a null list of {@link ByteBuffer}.
+   * 3. Create with an empty {@link ByteBuffer} list.
+   * 4. Create with a list of {@link ByteBuffer} containing a null {@link ByteBuffer}.
+   */
+  @Test
+  public void cornerCasesTest() {
+    // null byteBuffer
+    ByteBufferReadableStreamChannel readableStreamChannel;
+    try {
+      ByteBuffer content = null;
+      readableStreamChannel = new ByteBufferReadableStreamChannel(content);
+    } catch (RuntimeException e) {
+      Exception exception = (Exception) Utils.getRootCause(e);
+      assertTrue("Exception is not NullPointerException", exception instanceof NullPointerException);
+    }
+
+    // null bufferList
+    List<ByteBuffer> bufferList;
+    try {
+      bufferList = null;
+      readableStreamChannel = new ByteBufferReadableStreamChannel(bufferList);
+    } catch (RuntimeException e) {
+      Exception exception = (Exception) Utils.getRootCause(e);
+      assertTrue("Exception is not IllegalArgumentException", exception instanceof IllegalArgumentException);
+    }
+
+    // empty bufferList
+    try {
+      bufferList = Collections.EMPTY_LIST;
+      readableStreamChannel = new ByteBufferReadableStreamChannel(bufferList);
+    } catch (RuntimeException e) {
+      Exception exception = (Exception) Utils.getRootCause(e);
+      assertTrue("Exception is not IllegalArgumentException", exception instanceof IllegalArgumentException);
+    }
+
+    // bufferList containing a null byteBuffer
+    try {
+      ByteBuffer content = ByteBuffer.wrap(fillRandomBytes(new byte[32]));
+      bufferList = fillBufferList(content, null);
+      Collections.shuffle(bufferList);
+      readableStreamChannel = new ByteBufferReadableStreamChannel(bufferList);
+    } catch (RuntimeException e) {
+      Exception exception = (Exception) Utils.getRootCause(e);
+      assertTrue("Exception is not NullPointerException", exception instanceof NullPointerException);
+    }
+  }
+
   // helpers
   // general
 
   /**
+   * Create a list of {@code ByteBuffer} from given varargs.
+   * @param contents the variable {@code ByteBuffer} argument.
+   * @return list of {@code ByteBuffer}.
+   */
+  private List<ByteBuffer> fillBufferList(Object... contents) {
```

Comment:
wouldn't `Arrays.asList()` or `new ArrayList(Arrays.asList())` serve the same purpose as this function? Even if they don't allow `null` values, it seems like overkill to have a function for a single use case.